packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -1357,7 +1358,55 @@ test.serial('writeGranuleFromApi() writes a granule to PostgreSQL, DynamoDB, and
   t.true(await esGranulesClient.exists(granuleId));
 });
 
-test.serial('writeGranuleFromApi() writes a granule without an execution to PostgreSQL and DynamoDB.', async (t) => {
+test.serial('writeGranuleFromApi() given a partial granule updates only provided fields in PostgreSQL, DynamoDB, and Elasticsearch.', async (t) => {
+  const {
+    collectionCumulusId,
+    esClient,
+    esGranulesClient,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    knex,
+  } = t.context;
+
+  await writeGranuleFromApi({ ...granule }, knex, esClient, 'Create');
+
+  t.true(await granuleModel.exists({ granuleId }));
+  t.true(await granulePgModel.exists(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  ));
+  t.true(await esGranulesClient.exists(granuleId));
+
+  // Update existing granule with a partial granule object
+  const updatedGranule = {
+    granuleId,
+    collectionId: granule.collectionId,
+    cmrLink: 'updatedGranuled.com', // Only field we're changing
+    // FUTURE: In order to update a granule, the payload must include status and
+    // the status must be 'completed' or 'failed'
+    // if it's running or queued, it will try to insert the granule, not upsert
```

Comment:
This is probably something to address in https://bugs.earthdata.nasa.gov/browse/CUMULUS-2857. OR we document that `status` must always be provided in the payload, which doesn't seem as desirable but would preserve existing logic.

Reply:
Yes, that's indeed the problem I'm concerned with.

I ran into this very same issue with the rules endpoint quite a while back, where I was unable to remove meta properties from a rule because all rule updates (PUTs) are treated as "patches", not as replacements. I thought I had submitted a bug for this, but after searching Jira, I don't see such a bug reported, so I don't know if this was ever addressed.

Reply:
Sorry @chuckwondo I think we added comments at the same time. Yes, the PUT/PATCH confusion is problematic. See the Jira comment: https://bugs.earthdata.nasa.gov/browse/CUMULUS-3024

Reply:
@npauzenga  2857/follow on ticket are intended to address PUT/PATCH in phase 3.     The minimum for P3 is going to be to update the existing logic to allow for updates *regardless* of status/etc.      

  There's an implied limitation in how we implemented the user API in that it follows the same write rule logic as the messages, and I believe the approach was probably sub-optimal as Core wasn't originally implemented with the intent of direct user updates.        

In either case we have established work to address this, or will coming out of 2857, addressing the concern solidly falls out of scope for this ticket as we're trying to address the inconsistency here.